typing/parmatch.ml

```diff
@@ -431,10 +431,16 @@ let simple_match d h =
   | Constant c1, Constant c2 -> const_compare c1 c2 = 0
   | Lazy, Lazy -> true
   | Record _, Record _ -> true
+  | Record_unboxed_product _, Record_unboxed_product _ -> true
   | Tuple len1, Tuple len2 -> len1 = len2
```

Comment:
We check labels in the `Unboxed_tuple` case but use poly equal in this case.

Reply:
Indeed. I didn't want to duplicate the sad use of poly equal, and also didn't want to create unnecessary diffs with upstream by changing the existing code.  Though I guess the latter is pretty thinly motivated when we've changed the line before and the line after, and when the thing called `len` has had its type silently changed, so I'll push a similar improvement to the legacy code in a minute.  Thanks.